napari/layers/vectors/vectors.py

```diff
@@ -588,7 +626,12 @@ def _view_face_color(self) -> np.ndarray:
         """(Mx4) np.ndarray : colors for the M in view vectors"""
         face_color = self.edge_color[self._view_indices]
         face_color[:, -1] *= self._view_alphas
-        face_color = np.repeat(face_color, 2, axis=0)
+
+        if self.vector_style == 'line':
+            face_color = np.repeat(face_color, 2, axis=0)
+
+        elif self.vector_style == 'arrow':
+            face_color = np.repeat(face_color, 3, axis=0)
```

Comment:
Please make a test that creates a vector layer and changes the style to each of the valid styles. That should cover many lines of code and make sure nothing explodes in the future 😂

Reply:
I am looking into implement a test in `napari/_qt/_tests/test_qt_viewer.py` as you suggested in your original message.
Would something like that be enough ?
```python
@pytest.mark.parametrize('dim', [2, 3])
def test_vector_style_change(make_napari_viewer, dim):
    viewer = make_napari_viewer(ndisplay=dim)

    vector_data = np.array([[[0] * dim, [1] * dim]])
    vector_layer = viewer.add_vectors(vector_data)

    vector_layer.vector_style = 'line'
    vector_layer.vector_style = 'triangle'
    vector_layer.vector_style = 'arrow'
```

Reply:
@brisvag Do you know if the test above covers the situation described by @jni above ?

Reply:
I think that works @jules-vanaret, sorry for the silence. I'm investigating the test failures now...

Reply:
LOL It *was* enough, but instead of using `make_napari_viewer` you used `viewer = napari.Viewer()` in your test and that was the failure. 😅 See d65a030. @Czaki it's only because both CI and local tests were failing in the same weird way that I noticed this — I wonder whether we could have a CI or pre-commit check that ensures that `Viewer` doesn't get directly instantiated in tests?